packages/next/build/swc/src/next_dynamic.rs

```diff
@@ -0,0 +1,174 @@
+use swc_atoms::{js_word, JsWord};
+use swc_common::{FileName, DUMMY_SP};
+use swc_ecmascript::ast::{
+  ArrayLit, ArrowExpr, BinExpr, BinaryOp, BlockStmtOrExpr, CallExpr, Expr, ExprOrSpread,
+  ExprOrSuper, Ident, ImportDecl, ImportSpecifier, KeyValueProp, Lit, MemberExpr, ObjectLit, Prop,
+  PropName, PropOrSpread, Str, StrKind,
+};
+use swc_ecmascript::visit::Fold;
+
+pub fn next_dynamic(filename: FileName) -> impl Fold {
+  NextDynamicPatcher {
+    filename,
+    dynamic_binding: js_word!(""),
+  }
+}
+
+#[derive(Debug)]
+struct NextDynamicPatcher {
+  filename: FileName,
+  dynamic_binding: JsWord,
+}
+
+impl Fold for NextDynamicPatcher {
+  fn fold_import_decl(&mut self, decl: ImportDecl) -> ImportDecl {
+    let ImportDecl {
+      ref src,
+      ref specifiers,
+      ..
+    } = decl;
+    if &src.value == "next/dynamic" {
+      for specifier in specifiers {
+        if let ImportSpecifier::Default(default_specifier) = specifier {
+          self.dynamic_binding = default_specifier.local.sym.clone();
```

Comment:
This can be problematic if `next/dynamic` is imported multiple time.

Reply:
Also, this will fail for `import { default as foo } from 'next/dynamic'`. (I don't think it's important, though)